packages/react-components/src/components/chart/utils/getInfo.ts

```diff
@@ -0,0 +1,68 @@
+import { DurationViewport, Viewport } from '@iot-app-kit/core/src';
+import { v4 as uuid } from 'uuid';
+import { DEFAULT_MARGIN, tooltipLineColor, tooltipLineWidth } from '../eChartsConstants';
+import { ElementEvent } from 'echarts';
+import { SizeConfig } from '../types';
+export const isDurationViewport = (viewport: Viewport): viewport is DurationViewport =>
+  (viewport as DurationViewport).duration !== undefined;
+
+// TODO: test this once echarts live mode is supported
+export const viewportToSeconds = (viewport?: Viewport) => {
+  if (viewport && isDurationViewport(viewport)) {
+    switch (viewport?.duration) {
+      case '1m':
+        return { widthInMs: 60 * 1000, initial: Date.now() - 60 * 1000 };
```

Comment:
I think we use this parseDuration library already in this package. Might help clean this up